File: mikey/Java/OnlineVideoStore/java/src/main/java/Customer.java
import java.util.Vector;
import java.util.Enumeration;

public class Customer {
    public Customer(String name) {
        this.name = name;
    }

    public void addRental(Rental rental) {
        rentals.addElement(rental);
    }

    public String getName() {
        return name;
    }

    public String statement() {
        double totalAmount = 0;
        int frequentRenterPoints = 0;
        Enumeration rentals = this.rentals.elements();
        String result = "Rental Record for " + getName() + "\n";

        while (rentals.hasMoreElements()) {
            double thisAmount = 0;
            Rental rental = (Rental) rentals.nextElement();

            // determines the amount for rental line
            int daysRented = rental.getDaysRented();

            Movie movie = rental.getMovie();

            thisAmount = movie.determineAmount(thisAmount, daysRented);

            frequentRenterPoints++;

            if (movie.getPriceCode().getPriceCode() == Movie.NEW_RELEASE
                    && daysRented > 1)
                frequentRenterPoints++;

            result += "\t" + movie.getTitle() + "\t"
                    + String.valueOf(thisAmount) + "\n";
            totalAmount += thisAmount;

        }

        result += "You owed " + String.valueOf(totalAmount) + "\n";
        result += "You earned " + String.valueOf(frequentRenterPoints) + " frequent renter points\n";


        return result;
    }


    private String name;
    private Vector rentals = new Vector();
}

File: mikey/Python/OnlineVideoStorePy/Customer.py
from Movie import Movie


class Customer:
    def __init__(self, name):
        self.name = name
        self.rentals = []

    def add_rental(self, rental):
        self.rentals.append(rental)

    def get_name(self):
        return self.name

    def statement(self):
        totalAmount = 0.0
        frequentRenterPoints = 0
        rentals = self.rentals
        result = "Rental Record for " + self.get_name() + "\n"

        for each in rentals:
            thisAmount = 0.0
            if each.get_movie().get_price_code() == Movie.REGULAR:
                thisAmount += 2
                if each.get_days_rented() > 2:
                    thisAmount += (each.get_days_rented() - 2) * 1.5
            elif each.get_movie().get_price_code() == Movie.NEW_RELEASE:
                thisAmount += each.get_days_rented() * 3
            elif each.get_movie().get_price_code() == Movie.CHILDRENS:
                thisAmount += 1.5
                if each.get_days_rented() > 3:
                    thisAmount += (each.get_days_rented() - 3) * 1.5

            frequentRenterPoints += 1

            if each.get_movie().get_price_code() == Movie.NEW_RELEASE and each.get_days_rented() > 1:
                frequentRenterPoints += 1

            result += "\t" + each.get_movie().get_title() + "\t" + str(thisAmount) + "\n";
            totalAmount += thisAmount

        result += "You owed " + str(totalAmount) + "\n"
        result += "You earned " + str(frequentRenterPoints) + " frequent renter points\n"

        return result
